src/console_cmds.cpp

```diff
@@ -1227,7 +1227,12 @@ DEF_CONSOLE_CMD(ConRestart)
 	/* Don't copy the _newgame pointers to the real pointers, so call SwitchToMode directly */
 	_settings_game.game_creation.map_x = Map::LogX();
 	_settings_game.game_creation.map_y = Map::LogY();
-	_switch_mode = SM_RESTARTGAME;
+	if (_file_to_saveload.abstract_ftype == FT_SAVEGAME || _file_to_saveload.abstract_ftype == FT_SCENARIO || _file_to_saveload.abstract_ftype == FT_HEIGHTMAP) {
```

Comment:
please use `abstract_ftype != FT_NONE`. More future-proof.

Reply:
What about `FT_INVALID` which pop ups sometimes.

Reply:
Fair point. Should never make it in here, but better be safe than sorry. So yeah, != NONE && != INVALID sounds good!